text/0000-single-file-components.md

```diff
@@ -0,0 +1,111 @@
+  cart: inject()
+
+  passedItemPrice: 0,
+
+  layout: hbs`
+    <div>
+      <span>Item cost is: {{total}}</span>
+      <button {{action 'removeFromCart' target=cart}}>Remove</button>
+    </div>
+  `
+})
+```
+
+For proper syntax highlighting of handlebars inside the tagged template literal our
+team has created highlighters for
+[Sublime](https://github.com/healthsparq/sublime-ember-syntax),
+[Atom](https://atom.io/packages/language-ember), and
+[VSCode](https://marketplace.visualstudio.com/items?itemName=dhedgecock.ember-syntax).
+
+A proposed change to this pattern is allowing the component template to be declared
+using property `template` instead of `layout`, as it would likely be easier for newer
+users to remember.
+
+According to Robert Jackson, the existing pattern should be ok in production, and
+hopefully still is after the Glimmer 2 rewrites (we are using it in production at
+HealthSparq with no issues currently).
+Quoted: [Is it okay to use for components in production?](https://github.com/ember-cli/ember-cli-htmlbars-inline-precompile/issues/26)
+
+Work needed would be:
+- Updating Ember guides documentation for declaring component templates.
+- Creating a new component blueprint for inline template components.
+
+# How We Teach This
+
+Our team has been referring to this pattern as 'inline template declaration'. The
+syntax of declaring a template using a tagged template string can be documented in
+the Templates section of the Ember guides, perhaps in a new section called 'Declaring
+Templates Inline'. It could be presented as an alternate to defining a component's
+template using a separate .hbs file.
+
+Although this would change learning Ember as a new user, I personally think that it
+would make learning Ember easier if it is used as the primary pattern for creating
+components. Although the component generator creates the component and template files
+in the correct file locations, it may not be immediately clear to a new user that
+the file structure is required, and why the structure is required. Assigning a
+template to the `layout` (or `template`) property doesn't require understanding file
+lookups based on directory location.
+
+# Drawbacks
+
+1. In terms of teaching Ember, some might think that this complicates learning Ember
+  by giving new users two choices for how to declare a component's template.
+2. Although syntax highlighting for Sublime/Atom/VSCode exists, users must install
+  a package/extension to get highlighting.
+
+# Alternatives
+
+1. Officially not support single file component/template definitions.
+2. Support single file component/template definitions with a pattern different from
+  the tagged template string that works today.
+
+#### Other framework solutions:
+- Templates declared using property `template` and strings (Vue).
+- Templates declared with JSX (React).
+
+#### Impact of not doing:
+Missed developer convenience where other frameworks offer single file
```

Comment:
I fully understand wanting to have some assurance that the pattern will be supported, however, it seems to be that the RFC is laying down the argument for something slightly different and more involved.
I would say that as long as `layout` is a non-deprecated, public property, that you will be able to do this. Worst case scenario would be maintaining the precompiler yourselves, which, Ember already has a precompiler for templates.